### PR TITLE
Fix lots of small nits in sc-network

### PR DESCRIPTION
**Pull request by tomaka**
We do a lot of spurious cloning of `PeerId`s in `sc-network`. This cleans it up a bit, plus some other clean-ups.

Doesn't change any logic.

### CODE REVIEW

File: client/network/src/protocol/sync/blocks.rs [hunk 1]
```diff
@@ -185,26 +185,23 @@ impl<B: BlockT> BlockCollection<B> {
...
-						true
-					},
-					_ => {
-						false
-					}
-				};
-				if remove {
-					self.blocks.remove(&start);
+		if let Entry::Occupied(entry) = self.peer_requests.entry(who.clone()) {
+			let start = entry.remove();
```
Comment (arkpar) [inline client/network/src/protocol/sync/blocks.rs:189]:
```suggestion
		if let Some(start) = self.peer_requests.remove(who) {
```

File: client/network/src/protocol/generic_proto/handler/group.rs [hunk 2]
```diff
@@ -483,35 +483,35 @@ impl ProtocolsHandler for NotifsHandler {
 	) -> Poll<
 		ProtocolsHandlerEvent<Self::OutboundProtocol, Self::OutboundOpenInfo, Self::OutEvent, Self::Error>
 	> {
-		while let Poll::Ready(ev) = self.legacy.poll(cx) {
-			match ev {
+		if let Poll::Ready(ev) = self.legacy.poll(cx) {
```
Comment (andresilva) [inline client/network/src/protocol/generic_proto/handler/group.rs:486]: Was this change on purpose?

Comment (tomaka) [reply client/network/src/protocol/generic_proto/handler/group.rs:486]: Yes. It doesn't make sense to use a loop considering that all the code paths after entering the loop lead to a `return`. I moved the `return` right below the `if` to make that obvious.
